### PR TITLE
Hadoop enable blocksize variability and print performance numbers

### PR DESCRIPTION
**Pull request by gadgilrajeev**
Add ability to change block size in Hadoop Benchmark. 
Add ability to display performance in Hadoop Benchmark

### CODE REVIEW

File: perfkitbenchmarker/linux_packages/hadoop.py [hunk 1]
```diff
@@ -40,9 +40,13 @@
 _URL_OVERRIDE = flags.DEFINE_string(
     'hadoop_bin_url', None, 'Specify to override url from HADOOP_URL_BASE.')
 
+flags.DEFINE_integer('hadoop_blocksize', 256, 'blocksize in mb')
```
Comment (dorellang) [inline perfkitbenchmarker/linux_packages/hadoop.py:43]: Expand the description. Get some inspiration from: 
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_benchmarks/dpb_terasort_benchmark.py#L85

Save flag definition to a variable and access with it(like we are doing with _URL_OVERRIDE).

Also the default before was 128MB, please leave as it was.

**Review by dorellang — CHANGES_REQUESTED** [review]
Please sign the CLA.

File: perfkitbenchmarker/linux_packages/hadoop.py (hunk 1 above)
Comment (dorellang) [reply perfkitbenchmarker/linux_packages/hadoop.py:43]: Friendly reminder to address this comments.

**Comment by gadgilrajeev** [comment]
Yes, I will do so by the end of this week. Have been waiting to get
clearance to sign the CLA through my organization.

Thanks
Rajeev

**Comment by gadgilrajeev** [comment]
@dorellang

I am not sure why the pipeline fails. I have committed all changes to this
commit now according to your suggestions.

Thanks
Rajeev

File: perfkitbenchmarker/linux_packages/hadoop.py [hunk 2]
```diff
@@ -200,6 +205,8 @@ def _RenderConfig(vm,
   # This determines the number of reduce tasks in Terasort and is critical to
   # scale with the cluster.
   num_reduce_tasks = reduces_per_node * num_workers
+  if _BLOCKSIZE.value():
```
Comment (dorellang) [inline perfkitbenchmarker/linux_packages/hadoop.py:208]: This is should just be `.value:` (no parentheses).

**Review by dorellang — CHANGES_REQUESTED** [review]
Could you provide in the comments the logs of a successful execution? (also including the command you ran, so I can reproduce it)

File: perfkitbenchmarker/linux_packages/hadoop.py (hunk 2 above)
Comment (gadgilrajeev) [reply perfkitbenchmarker/linux_packages/hadoop.py]: Resolved

File: perfkitbenchmarker/linux_packages/hadoop.py [hunk 3]
```diff
@@ -200,6 +205,8 @@ def _RenderConfig(vm,
   # This determines the number of reduce tasks in Terasort and is critical to
   # scale with the cluster.
   num_reduce_tasks = reduces_per_node * num_workers
+  if _BLOCKSIZE_OVERRIDE.value:
+    block_size = _BLOCKSIZE_OVERRIDE * 1024 * 1024
```
Comment (dorellang) [inline perfkitbenchmarker/linux_packages/hadoop.py:209]:
```suggestion
    block_size = _BLOCKSIZE_OVERRIDE.value * 1024 * 1024
```

**Review by dorellang — CHANGES_REQUESTED** [review]

**Review by dorellang — APPROVED** [review]

**Comment by dorellang** [comment]
Can you remove the 'Co-authored-by:' lines from this branch commit messages with `git rebase -i master`? The CLA checker bot doesn't like the fact my personal email is showing up there.

**Comment by bvliu** [comment]
You may need to sync past https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/commit/c6bc0f65c6a493ceb8c3be260d0acdad1672cda4 for the build to work.

**Comment by bvliu** [comment]
Your cloudbuild.yaml file is out of date, please sync/rebase past https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/commit/c6bc0f65c6a493ceb8c3be260d0acdad1672cda4.

**Comment by gadgilrajeev** [comment]
[00980d1d_pkb.log](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/files/10745443/00980d1d_pkb.log)

**Comment by bvliu** [comment]
Thanks!

**Review by dorellang — APPROVED** [review]

**Comment by pmkc** [comment]
I merged the flag as `--hadoop_hdfs_blocksize`. I didn't merge the changes to hadoop_terasort_benchmark.py, because it is no longer maintained and I would recommend you look at dpb_terasort_benchmark.py, which [has data size in the metadata](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_benchmarks/dpb_terasort_benchmark.py#L182)

We should formally deprecate the spark_service API and benchmarks. Sorry about that.

I should have commented on your review instead of changing it as I merged, but I didn't catch it in time. Let me know if this works for you.

**Comment by gadgilrajeev** [comment]
Thanks for pointing that out to me. This should work for me.

-Rajeev

**Pull request merged**